src/diffenator2/__init__.py

```diff
@@ -94,20 +105,20 @@ def ninja_diff(
             diffenator,
             out,
             imgs,
+            styles,
             user_wordlist,
             filter_styles,
             pt_size,
             threshold=threshold,
         )
         return
-    styles_before = styles_in_fonts(fonts_before)
-    styles_after = [style for _, style, _ in styles_in_fonts(fonts_after)]
-    styles = [s for s in styles_before if s[1] in styles_after]
-    partitioned = partition(styles, MAX_STYLES)
-    if not os.path.exists(out):
-        os.mkdir(out)
+
+    matcher = FontMatcher(fonts_before, fonts_after)
+    getattr(matcher, styles)()
```

Comment:
I'd appreciate a check here to make sure that `old_styles` and `new_styles` are not empty, and giving a sensible errors message if they are.

Reply:
Will now raise a ValueError with the message:

```PYTHON
Traceback (most recent call last):
  File "/Users/marcfoley/Type/diffenator2/venv/bin/diffenator2", line 8, in <module>
    sys.exit(main())
  File "/Users/marcfoley/Type/diffenator2/src/diffenator2/__main__.py", line 62, in main
    ninja_diff(
  File "/Users/marcfoley/Type/diffenator2/src/diffenator2/__init__.py", line 119, in ninja_diff
    raise ValueError(
ValueError: Matcher was not able to detect any matching styles for instances method.
Please ensure that variable fonts have fvar instances, both fonts have designspaces which overlap or ensure that both sets of static fonts have some matching styles.
```